reference-implementation/lib/helpers.js

```diff
@@ -62,6 +62,10 @@ exports.IsFiniteNonNegativeNumber = v => {
 };
 
 exports.InvokeOrNoop = (O, P, args) => {
+  assert(exports.typeIsObject(O) === true);
```

Comment:
It's actually a bit weirder. For example this code will work:

```js
Number.prototype.start = controller => { ... };
new ReadableStream(5);
```

This is purely for consistency with existing platform APIs/JavaScript destructuring.